SwiftCompilerSources/Sources/Optimizer/Utilities/WalkUtils.swift

```diff
@@ -69,6 +69,43 @@ protocol WalkingPath : Equatable {
 
 extension SmallProjectionPath : WalkingPath { }
 
+/// A `WalkingPath` where `push` and `pop` instructions
+/// are forwarded to an underlying `projectionPath`.
+protocol SimpleWalkingPath : WalkingPath {
```

Comment:
`SimpleWalkingPath` is not a good name, IMO. How about `SmallProjectionWalkingPath`?